cmd/duffle/main.go

```diff
@@ -130,17 +131,38 @@ type driverWithRelocationMapping struct {
 }
 
 func (d *driverWithRelocationMapping) Run(op *driver.Operation) error {
-	// if there is a relocation mapping, ensure it is mounted
+	// if there is a relocation mapping, ensure it is mounted and relocate the invocation image
 	if d.relMapping != "" {
 		op.Files["/cnab/app/relocation-mapping.json"] = d.relMapping
+
+		if err := d.relocateImage(&op.Image); err != nil {
+			return err
+		}
 	}
+
 	return d.driver.Run(op)
 }
 
 func (d *driverWithRelocationMapping) Handles(it string) bool {
 	return d.driver.Handles(it)
 }
 
+func (d *driverWithRelocationMapping) relocateImage(im *string) error {
```

Comment:
I guess it's just a matter of taste, but what do you think if we could avoid passing mutable parameters, and return the new relocated image instead?
```go
func (d *driverWithRelocationMapping) relocateImage(im string) (string, error) {
```

Reply:
Happy to oblige. Pushed a fix.